autometa/common/external/samtools.py

```diff
@@ -68,7 +68,12 @@ def sort(sam, bam, cpus=mp.cpu_count()):
         cmd = f"samtools view -@{cpus} -bS {sam} | samtools sort -@{cpus} -o {temp_bam}"
         logger.debug(f"cmd: {cmd}")
         with open(out, "w") as stdout, open(err, "w") as stderr:
-            subprocess.run(cmd, stdout=stdout, stderr=stderr, shell=True, check=True)
+            proc = subprocess.run(cmd, stdout=stdout, stderr=stderr, shell=True)
+            try:
+                proc.check_returncode()
+            except subprocess.CalledProcessError as err:
+                print(f"Error while running samtools sort \n cmd: {cmd}")
```

Comment:
We should not be printing here. We should be using the `logger`.

```suggestion
                logger.error(f"SamtoolsSortError: {cmd}")
```

Reply:
Ultimately, since we will know where the failure occurs, we could log the error and handle the error where the function is called. We could similarly put a SamtoolsSortError in exceptions.py to handle this